### PR TITLE
[RPHAM-3106] Getting error then try to generate classes from WSDL file or URL

### PR DESCRIPTION
**Pull request by fjtirado**
Generating custom class loader with list of jars in case ModuleClassLoader is found in hierachy

**JIRA**:  [RHPAM-3106](https://issues.redhat.com/browse/RHPAM-3106)

### CODE REVIEW

**Comment by mareknovotny** [comment]
Jenkins do fdb

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (elguardian) [inline jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:566]: bad naming. this is a classpath phisical location. hackUrls says very little. 
urlLocationClassPathLibs or so

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (elguardian) [inline jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:564]: says very little. You are bypassing the class loader problem of CXF. what about
CXFJavaCompilerClassLoader or so ?

**Comment by elguardian** [comment]
Good work in general @fjtirado. Just try to be meaningful with the naming if anything.
This was a good effort of research, test and proper fix. Keep it up.
We can defer CXF fix. With this we have enough.

**Comment by fjtirado** [comment]
Adding comments and renaming fields

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java]: Thanks for the suggestion, I was unable to find a proper name ;)

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java]: yes, I changed to jarUrls, which is shorter that urlLocationClasspathLibs and more accurate than hackURLs

**Review by elguardian — APPROVED** [review]
good to go from eng side.

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (afalhambra) [inline jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:627]: Maybe a more descriptive method name can be useful? something like `getFieldObject`?

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (afalhambra) [inline jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:573]: Yep, as per comment, maybe it's worth filtering filenames with "jaxb"

**Review by afalhambra — COMMENTED** [review]
Only minor comments. Overall looks pretty good

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:551]: I prefer to reduce method invocations as possible. 
Also, if in future we support more classloaders we will extend that if statement here, so it makes sense to keep it where it is

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:555]: There is not need to log (this is reponsibility of exceptin handler in upper layers) but it seems more correct to replace IllegalStateException by WorkItemRuntimeException

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:562]: if uris is empty, it means there are not ModuleClassLoader in hierarchy and we can safely use the "standard" classloader

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:585]: same as above

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:559]: we should not assume anything about classloader hierarchy, this code will be executed in other containers different than wildfly

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:593]: Implementation details should not be part of java doc.
See this old but still aplicable link https://www.oracle.com/technetwork/java/codeconventions-150003.pdf (section 5.1.1)

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:627]: sure, will change

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (fjtirado) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:573]: this is a comment to easily locate the piece of code to change if, in future, we need to generate a smaller classpath for compiler.

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (elguardian) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:559]: Actually it is a bit more complex that that. you can have multiple parents in here. but this should be enough. We can argue that we should stopped looking after finding the wildfly module class loader. I leave that to QE to decide

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (elguardian) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:573]: it should be all of them. That comment should be removed. Please try to avoid this sort of comments. If you want to point out something to reviewers just add a comment in github pull request.

File: jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java
Comment (elguardian) [reply jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java:551]: we won't support any class loaders. This is a temporary solution that should go to some point to cxf so this will be lost at some point.

**Review by afalhambra — APPROVED** [review]
Very good work @fjtirado!

**Comment by fjtirado** [comment]
@kiegroup/gatekeepers please merge this

**Comment by ricardozanini** [comment]
I would merge it, but we have CI errors. Have you guys taken a loot at it?

**Comment by fjtirado** [comment]
> I would merge it, but we have CI errors. Have you guys taken a look at it?
Looks unrelated, restarting build to see if we got green light, thanks for pointing it out

**Comment by mareknovotny** [comment]
@fjtirado it is more than recommended to run FDB here on jbpm repo as droolsjbpm-integration is in bad status always we change something in drools or Jbpm and FDB is not checked.

Jenkins do fdb

**Comment by fjtirado** [comment]
@mareknovotny FDB fails with unrelated test

**Review by gmunozfe — APPROVED** [review]
Looks good to me, awesome work @fjtirado !

**Comment by ricardozanini** [comment]
@mareknovotny it's up to you to merge, I prefer to stay out since I'm not familiar with this CI process nor with the changes in this PR.

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Review by mareknovotny — APPROVED** [review]
```
all works on my local env too.
Thanks @elguardian 
```
wrong browser tab :(, please ignor this

**Comment by fjtirado** [comment]
> Jenkins do fdb

Still failing in unrelated test

**Comment by mareknovotny** [comment]
> > Jenkins do fdb
> 
> Still failing in unrelated test

I think it is ok to merge when jira is approved for the target release ;)
The failed tests are already reported as https://issues.redhat.com/browse/JBPM-9383

**Comment by ricardozanini** [comment]
> I think it is ok to merge when jira is approved for the target release ;)

Just let me know.

**Comment by fjtirado** [comment]
> > I think it is ok to merge when jira is approved for the target release ;)
> 
> Just let me know.

I think all acks has been added to JIRA now, so please free to proceed

**Pull request merged**
